app/controllers/tags_controller.rb

```diff
@@ -57,10 +57,14 @@ def set_instance_presenter
 
   def collection_presenter
     ActivityPub::CollectionPresenter.new(
-      id: tag_url(@tag, params.slice(:any, :all, :none)),
+      id: tag_url(@tag, filter_params),
       type: :ordered,
       size: @tag.statuses.count,
       items: @statuses.map { |s| ActivityPub::TagManager.instance.uri_for(s) }
     )
   end
+
+  def filter_params
+    params.slice(:any, :all, :none).permit(:any, :all, :none)
```

Comment:
Why use both `slice` and `permit`? Isn't `permit` enough?

Reply:
Permit isn't enough when you want to convert to hash and the hash contains params that are not permitted.

Reply:
As far as I understand it, both `slice` and `permit` return an `ActionController::Parameters`, sliced to the params, with `permit` marking them as permitted while `slice` doesn't. So `slice` alone isn't good for that use case, but `permit` is?

Reply:
```
[1] pry(main)> p = ActionController::Parameters.new(foo: '1', bar: '2')
=> <ActionController::Parameters {"foo"=>"1", "bar"=>"2"} permitted: false>
[2] pry(main)> p.permit(:foo).to_h
Unpermitted parameter: :bar
=> {"foo"=>"1"}```

Reply:
for some reason i can't reproduce this here, but ok